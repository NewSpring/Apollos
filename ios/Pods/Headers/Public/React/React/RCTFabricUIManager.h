../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h