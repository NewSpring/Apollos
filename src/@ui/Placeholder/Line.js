import React from 'react';
import { View } from 'react-native';
import Placeholder from 'rn-placeholder';
import styled from '@ui/styled';

export const Line = styled(({ theme, width = '100%' }) => ({
  borderRadius: theme.sizing.borderRadius,
  backgroundColor: theme.colors.background.inactive,
  height: theme.helpers.rem(1),
  width,
  alignSelf: 'stretch',
}), 'Placeholder.Line')(({ style }) => (
  <View style={style} />
));

export default Placeholder.connect(Line);
