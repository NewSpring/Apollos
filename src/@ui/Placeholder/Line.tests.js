import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';
import { Line } from './Line';

describe('The Line placeholder', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Providers><Line /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('accepts width', () => {
    const tree = renderer.create(
      <Providers><Line width="50%" /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('accepts custom styles', () => {
    const tree = renderer.create(
      <Providers><Line width="50%" style={{ backgroundColor: 'salmon' }} /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
