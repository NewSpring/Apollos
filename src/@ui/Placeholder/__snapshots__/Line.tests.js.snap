// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Line placeholder accepts custom styles 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "salmon",
      "borderRadius": 6,
      "height": 18,
      "width": "50%",
    }
  }
/>
`;

exports[`The Line placeholder accepts width 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#dddddd",
      "borderRadius": 6,
      "height": 18,
      "width": "50%",
    }
  }
/>
`;

exports[`The Line placeholder renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#dddddd",
      "borderRadius": 6,
      "height": 18,
      "width": "100%",
    }
  }
/>
`;
